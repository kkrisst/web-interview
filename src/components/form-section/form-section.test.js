import React from 'react'
import { shallow } from 'enzyme'

import FormSection from './form-section.component.jsx'

describe('Testing the FormSection component', () => {
  let wrapper

  beforeEach(() => {
    const props = {
      title: 'Consultant Type',
      iconName: 'stethoscope',
    }
    wrapper = shallow(<FormSection {...props} />)
  })

  it('expect FormSection to be defined', () => {
    expect(FormSection).toBeDefined()
  })

  it('expect to render FormSection component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has the correct title', () => {
    const titleElement = wrapper.find('.title')
    expect(titleElement.text()).toEqual('Consultant Type')
  })

  it('can have other title than Consultant Type', () => {
    const props = {
      title: 'alma',
      iconName: 'stethoscope',
    }
    const wrapper = shallow(<FormSection {...props} />)
    const titleElement = wrapper.find('.title')
    expect(titleElement.text()).toEqual('alma')
  })

  it('expext to pass the correct iconName', () => {
    const iconElement = wrapper.find('FontAwesomeIcon')
    expect(iconElement.props().icon).toEqual('stethoscope')
  })
})
