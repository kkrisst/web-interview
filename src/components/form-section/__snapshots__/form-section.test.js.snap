// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the FormSection component expect to render FormSection component 1`] = `ShallowWrapper {}`;
